flocker/control/httpapi.py

```diff
@@ -200,17 +200,97 @@ def create_dataset_configuration(self, primary, dataset_id=None,
         saving = self.persistence_service.save(new_deployment)
 
         def saved(ignored):
-            result = {
-                u"dataset_id": dataset_id,
-                u"primary": primary,
-                u"metadata": metadata,
-            }
-            if maximum_size is not None:
-                result[u"maximum_size"] = maximum_size
+            result = api_dataset_from_dataset_and_node(dataset, primary)
             return EndpointResponse(CREATED, result)
         saving.addCallback(saved)
         return saving
 
+    def _find_manifestation_and_node(self, dataset_id):
+        """
+        Given the ID of a dataset, find its primary manifestation and the node
+        it's one.
+
+        :param unicode dataset_id: The unique identifier of the dataset.  This
+            is a string giving a UUID (per RFC 4122).
+
+        :return: Tuple containing the primary ``Manifestation`` and the
+            ``Node`` it is on.
+        """
+        # Get the current configuration.
+        deployment = self.persistence_service.get()
+
+        # Lookup the node that has a primary Manifestation (if any)
+        manifestations_and_nodes = manifestations_from_deployment(
+            deployment, dataset_id)
```

Comment:
Deleting a deleted dataset should give same response, if we want this API to be idempotent (and we do, I think). So I think that's OK.

The second case is definitely a bug. I'll fix it when I get in.